MetaMask/metamask-mobile
# feat: cicd e2e label requirements + pr automation

## **Description**

This PR does a few things
1. Automatically set new PRs targeting main to "draft" state.
2. Enforce the E2E Bitrise Status Check on lifecycle on incoming PRs which will essentially enforce that the PR has 1 of the two labels 'Run Smoke E2E' or 'No E2E Smoke Needed'




With this change we're looking to improve the posture on our PR lifecycle with auto drafts and enforce a higher standard for incoming changes towards or E2E testing posture.

## **Related issues**

[Fixes:]

https://github.com/MetaMask/metamask-mobile/issues/11747

## **Manual testing steps**

UC1 : Nether E2E Label is set
-> https://github.com/MetaMask/metamask-mobile/actions/runs/11826040868/job/32951145968?pr=12243 
    Expected : Not Mergeable

UC2 : docs PR

Should run: false, Reason: The pull request is documentation related.
Skipping Bitrise status check. due to the following reason: The pull request is documentation related.
Created 'Bitrise E2E Status' check with skipped status for commit dea08a922c8a573e
https://github.com/MetaMask/metamask-mobile/actions/runs/11826187817/job/32951608300?pr=12243

Expected: Mergeable

UC3 : No E2E Smoke Needed Label is set

Anti label: true
Should run: false, Reason: The pull request has the anti-label.
Skipping Bitrise status check. due to the following reason: The pull request has the anti-label.
Created 'Bitrise E2E Status' check with skipped status for commit dea08a9[22]
https://github.com/MetaMask/metamask-mobile/actions/runs/11826234735/job/32951749305?pr=12243

Expected Mergeable


UC4: Smoke E2E Label is set

Should run: true, Reason: The smoke test label is present.
Starting Bitrise build for commit dea08a922c8a573e167a88f03e43a41ba25f24fd


## **Screenshots/Recordings**



### **Before**



N/A CICD Changes only

### **After**



N/A CICD Changes only

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
